src/placeos-frontend-loader/loader.cr

```diff
@@ -143,20 +143,20 @@ module PlaceOS::FrontendLoader
       # Grab commit for the cloned/pulled repository
       checked_out_commit = Git.current_repository_commit(repository.folder_name, content_directory)
 
-      # Update model commit if the repository is not held at HEAD
-      unless checked_out_commit == repository_commit
-        if repository_commit != "HEAD"
-          Log.info { {
-            message:           "updating commit on Repository document",
-            current_commit:    checked_out_commit,
-            repository_commit: repository_commit,
-            folder_name:       repository.folder_name,
-          } }
+      # Update model commit iff...
+      # - the repository is not held at HEAD
+      # - the commit has changed
+      unless repository_commit.starts_with?(checked_out_commit) || repository_commit == "HEAD"
```

Comment:
What's the expected contents of both `repository_commit` and `checked_out_commit`?

To remove potential for error here. Is it possible to enforce the commit stored against the repo model is the full 160 bit form? Alternatively, if there's a needed to supported truncated id's, could this be flipped so that `Git.current_repository_commit` provides that full hash and `#starts_with?` runs on that.

Reply:
The flip seems like the way to go.
That requires updates in another library, so I think this is fine to go if the other specs pass.
I'll do the git updates separately

Reply:
https://github.com/PlaceOS/compiler/pull/42